OpenRA.Game/Graphics/WorldRenderer.cs

```diff
@@ -153,14 +153,14 @@ void GenerateRenderables()
 		// PERF: Avoid LINQ.
 		void GenerateOverlayRenderables()
 		{
-			foreach (var a in World.ActorsWithTrait<IRenderAboveShroud>())
+			World.ApplyToActorsWithTrait<IRenderAboveShroud>((Actor actor, IRenderAboveShroud trait) =>
```

Comment:
Its not necessary to specify all three types here. This could be either
```
World.ApplyToActorsWithTrait<IRenderAnnotations>((actor, trait) =>
```
or
```
World.ApplyToActorsWithTrait>((Actor actor, IRenderAnnotations trait) =>
```

IMO the first is easier to read.